### ISSUE TITLE
Feature Request: Add Analysis Services Support

### ISSUE THREAD

**Issue by demyte** [Microsoft/sqlopsstudio#1026]
It would be great to be able to connect to Analysis Services instances (both multi-dimensional and tabular) and run MDX, DAX and XMLA queries.

**Comment by cwebbbi** [comment]
I absolutely want Analysis Services support too. I would love full support for all flavours of Analysis Services but in practical terms I would be happy with support for just Analysis Services Tabular 2016+ and Azure Analysis Services, and eventually Power BI Premium via XMLA endpoints (which should be available in 2019). For language support, as well as DAX I would also want M and TMSL too; MDX would be nice but is less of a priority. I would also like the ability to connect to Analysis Services xEvents.

**Comment by PaulTinNZ** [comment]
Second all that's been said and particularly connect to Analysis Services xEvents.
Would also think that the SQL Server Analysis support team at MS must have plenty of tools that would enhance our management and vision into the application.
Like showing which session is using all the available server memory...just one quick example.

**Comment by JanMulkens** [comment]
I'd love everything that's on Chris' wishlist! 🙂

https://blog.crossjoin.co.uk/2018/11/08/azure-data-studio-analysis-services-power-bi/

**Comment by matt40k** [comment]
It would also help with the cross platform goals

**Comment by otykier** [comment]
Yes! I believe a prerequisite for this, would be to compile the [AMO/TOM library](https://www.nuget.org/packages/Microsoft.AnalysisServices.retail.amd64/) against .NET Standard, right? That would also be of great value to the community - imagine, for example, a cross-platform version of [Tabular Editor](https://github.com/otykier/tabulareditor).

**Comment by GrosR** [comment]
yes, we all want this!

**Comment by erapade** [comment]
Agree. Would like to be able to write DAX queries kind of the same way (using the MDX mode) as in SSMS

**Comment by affsantos** [comment]
Please Microsoft, do this! Multidimensional and Tabular on-prem :)

**Comment by PietkaSmig** [comment]
Hi, what do you think when can you start working on this ? Basically the soft is pretty cool, works very fast and looks nice however any BI developer in an average size company is using SSAS and SQL. You prepare data in sql and prepare tabular model - the amount of data always put into this direction. Adding MySQL would be also a very good move. At least you could start from ODBC drivers and get closer to functionality available in MS Excel .

**Comment by AlejoSQL** [comment]
Absolutely! @vickyharp , this goes in line with the conversations about modernizing the experience for more traditional DB / no data science people. Support for MDX, DAX, TMSL and M would be a massive win!!!

**Comment by mmichtch** [comment]
It was a great disappointment when I found that SSAS is not supported. Please include this in future releases.

**Comment by StefanieCulley** [comment]
Another vote for adding the BI side of the house.  Please!

**Comment by kfrankie** [comment]
**Azure Data** Studio but without Azure Analysis Services ? please fix this inconsistency :)

**Comment by rafaelsmoreno** [comment]
Any news on that? Hope it happens soon!

**Comment by chappers2003** [comment]
Definitely needs analysis services, it does seem odd to go ahead without it. The number of people/businesses that this would help definitely needs to be a priority.

**Comment by oynek** [comment]
That would be a game changer for me.

**Comment by aisbergde** [comment]
one more argument: Azure Data Studio can be installed and used without Admin rights, SSMS can't

**Comment by HeidiHasting** [comment]
Agreed, plus 1 from me. If they could incorporate in like an MDXStudio style editor, results pane and format option that would be great.

**Comment by paulturley** [comment]
Rather than reinventing the wheel, take a very close look at community-developed applications like DAX Studio and Tabular Editor, and incorporate similar functionality. In the vacuum of modern support for SSAS/AAS/PBI, the community have developed some amazing tools. There are other good community tools but these are the most rich and recognizable. I would defer to Chris Webb to lead this charge as he has been on-point for many years, before and after joining the Microsoft CAT. I'm certain that a consensus from Chris, Christian Wade, Darren Gosbell, Marco Russo and Daniel Otykier would yield the best set of requirements.

**Comment by rgpdx** [comment]
Enabling Analysis services in Azure Data Studio is a must-have for me to use Azure Data Studio. Please consider this feature add.

**Comment by kevinobrien54** [comment]
Please add Analysis Services to Azure Data Studio

**Comment by GeorgeDom** [comment]
Please add Analysis Services to Azure Data Studio.

**Comment by matt40k** [comment]
Can folks do a thumbs up on the original post rather than just adding a comment saying "Please add Analysis Services to Azure Data Studio" - yes, we all want it, and it gets me all depressed when I see an update on this issue and it isn't progress it's just more people saying they really need it. </rant>

**Comment by Jcastro6** [comment]
Please add Analysis Services to Azure Data Studio.

**Comment by jgibbycsc** [comment]
For our team to use this tool, AAS support is needed. Thank you for your hard work Dev Team!!

**Comment by PietkaSmig** [comment]
Hi Guys, 
Any chance this will appear in the near feature ? ADS would become a standalone tool for data analysts after adding this feature - currently you need to have SSMS anyway.

**Comment by sdjdk** [comment]
Please add Analysis Services to Azure Data Studio

**Comment by nrichmond-fi** [comment]
need

**Comment by InathiM** [comment]
Please add Analysis Services capabilities to Azure Data Studio

**Comment by MikeCarlo** [comment]
I would also really like this feature. I would like to be able to add a profile for the Analysis services so I can watch how it incrementally refreshes data.

**Comment by maordahari** [comment]
Me also

**Comment by pdobrokhotov** [comment]
Agree! Connecting to SSAS Multidimensional and Tabular is an absolute MUST

**Comment by tmjones212** [comment]
This sure would be neat

**Comment by NickyvVr** [comment]
I can't believe this is not possible yet, please add support for it

**Comment by senkadir** [comment]
It's been four years since the ticket was created. This is shame. Please implement!

**Comment by PietkaSmig** [comment]
Definitely should be implemented also as an additional connection type in Jupyther notebook. It suppoused to be software for data analysts and we are missing DAX !

**Comment by lbendlin** [comment]
The latest versions of SSMS get worse and worse for Power BI dataset and partition management. It would be great if there was a viable alternative. Please support Analysis Services connections!

**Comment by kewlhand** [comment]
This is long overdue.

**Comment by SQLGator** [comment]
This is a great opportunity to add this feature to Azure Data Studio especially given that we converted our AS tabular models to Power BI Premium datasets and now we cannot use Visual Studio to deploy them as they do not support the compatibility level required. We have to use Tabular Editor so in effect there is no Microsoft product to deploy a Microsoft tabular model to a Microsoft Power BI premium dataset.Please help.

**Comment by yuranos** [comment]
Please, consider this seriously!

**Comment by vetleandreas** [comment]
Still no clear decision on this?

**Comment by rossgalb** [comment]
I am so surprised that this is not being developed yet! Anyone who manages a data warehouse using microsoft products would have both sql and AS would they not? Having to switch between azure data studio and ssms is very annoying.

**Comment by arkdvd** [comment]
This is much needed... any one could enable/disable an  tabular service support extension. Its cool to see in ADS

**Comment by Milhouse** [comment]
Yes, was just trying to connect to my Power BI XMLA endpoint server with Azure Data Studio to see if I could use the Profiler there instead of in SSMS. I know that Profiler is heavier on the server than xEvents, and would rather use those.
